File: 45-jump-game-ii/jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int left = 0;
        int right = 1;
        int steps=0;
        while(right<nums.size())
        {
            steps++;
            int val =0;
            for(int i=left;i<right && i<nums.size();i++)
            {
             val = max(val,nums[i]+i);
            }
            if(val>=nums.size()-1) return steps;
            left = right;
            right = val+1;
        }
        return steps;
    }
};

File: 45-jump-game-ii/jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
       int left =0;
       int right =0;
       int steps=0;
       int maxreach = 0;
       while(left < nums.length){
         
         for(int i=left;i<=right;i++){
            if(i==nums.length-1) return steps;
            maxreach = Math.max(maxreach,nums[i]+i);
         }
         steps++;
         left = right+1;
         right = maxreach;
       } 
       return steps;
    }
}